### DISCUSSION TITLE
Question: How to tell the program to download the highest resolution

### DISCUSSION THREAD

**Discussion by busdriverflix** [omansak/libvideo#207] [Q&A] [answered]
Hi, I'm new to programming and as a starter project I wanted to make a YouTube Downloader. I want to use this library, but I have one big problem. I just can't find how to select resolutions. In the document is described how to get info about a video, even the resolution, but not how to tell the programm to download the highest resolution (or I'm just blind and can't see it). I googled a lot to find a solution to this, but doesn't matter what I do, when I launch my program and start downloading a video, it's always just in 360p.
Please someone help me.

**Comment by omansak** [comment, marked as answer]
you can use like in this doc.

https://github.com/omansak/libvideo/blob/master/docs/README.md#advanced
```csharp
var youTube = YouTube.Default; // starting point for YouTube actions
var videoInfos = youTube.GetAllVideosAsync(link).GetAwaiter().GetResult();
var maxResolution = videoInfos.First(i => i.Resolution == videoInfos.Max(j => j.Resolution));
```

**Reply by busdriverflix** [reply]
So I tried using this, but no matter what I do, the resolution stays at 360p. Can you please help me again? I don't understand how I can set the resolution to the maximum. I tried this var maxResolution line but it just doesn't work. Is this how you set it to maximum resolution or do I have to change something and if yes what exactly?
What I learned is that the GetAllVideos() function returns an iEnumerable and I just have to iterate through it, and select the highest resolution, but I don't know how. I would really appreciate, if you could clarify

**Reply by marnagy** [reply]
@busdriverflix  Could you add an example URL? There is also a possibility that the video is uploaded in 360p to Youtube and thus there is no higher resolution available.
